### ISSUE TITLE
THREE.InstancedBufferGeometry, THREE.MultiMaterial and groups

### ISSUE THREAD

**Issue by arnaldof** [mrdoob/three.js#8912]
##### Description of the problem

I need some clarification regarding the use of `addGroup` method in `THREE.InstancedBufferGeometry`, when using it with `THREE.MultiMaterial`.

This is r75, source looks like this:

```
THREE.InstancedBufferGeometry.prototype.addGroup = function ( start, count, instances ) {
    this.groups.push( {
        start: start,
        count: count,
        instances: instances
    } );
};
```

I see no other use of field `instances` besides the "copy" method. There is no "materialIndex" field on this groups like in `THREE.BufferGeometry`.

If I try to render something like `Mesh( InstancedBufferGeometry, MultiMaterial )` does not work, I get an error from "visible" field in group material.

Replacing `instances` with `materialIndex` in `addGroup` works just fine, but must be another use intended for `instances` field. I'm not understanding this, or maybe this is a bug.

Original question in stackoverflow: http://stackoverflow.com/questions/37281166/three-instancedbuffergeometry-three-multimaterial-and-groups
##### Three.js version
- [x] r75
##### Browser
- [x] All of them

**Comment by benaadams** [comment]
Not sure as its all been heavily refactored; and the instancing was there before the multimaterial.

The instances are the count of instances you want of those attributes; though I'm guessing it should be `maxInstancedCount` but that may be picked up from the geometry itself, so it may be a typo and should be `materialIndex`?

**Comment by benaadams** [comment]
Looks like it currently calls into [WebGLIndexedBufferRenderer](https://github.com/mrdoob/three.js/blob/dev/src/renderers/webgl/WebGLIndexedBufferRenderer.js) ?

**Comment by Mugen87** [comment]
Should be solved via #9815
